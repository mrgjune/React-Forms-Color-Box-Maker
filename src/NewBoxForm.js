import React from 'react';
class NewBoxForm extends React.Component {
    constructor(props) {
        super(props);
        this.state = { width: 0, height: 0, color: "" }

        this.handleChange = this.handleChange.bind(this);
        this.handleSubmit = this.handleSubmit.bind(this);
    }
    /** Send {height, width, color} to parent
    *    & clear form. */
    handleSubmit(evt) {
        evt.preventDefault();
        this.props.addBox(this.state);
        this.setState({ width: 0, height: 0, color: "" });
    }

    /** Update local state w/curr state of input elem */

    handleChange(evt) {
        this.setState({
            [evt.target.name]: evt.target.value
        })
    }
    /** render form */
    render() {
        return (
            <form onSubmit={this.handleSubmit}>
                <label htmlFor="color">Background Color:</label>
                <input id="color"
                    name="color"
                    value={this.state.color}
                    onChange={this.handleChange} />

                <label htmlFor="width">Width:</label>
                <input type="number"
                    id="width"
                    name="width"
                    value={this.state.width}
                    onChange={this.handleChange} />

                <label htmlFor="height">Height:</label>
                <input type="number"
                    id="height"
                    name="height"
                    value={this.state.height}
                    onChange={this.handleChange} />

                <button>Add a new item!</button>
            </form>
        );
    }
}




export default NewBoxForm;