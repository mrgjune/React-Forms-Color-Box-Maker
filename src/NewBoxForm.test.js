import React from "react";
import { shallow, mount } from "enzyme";
import toJson from "enzyme-to-json";
import NewBoxForm from "./NewBoxForm";

it("renders without crashing", function () {
    shallow(<NewBoxForm />);
  });

// snapshot test
it("matches snapshot", function () {
  let wrapper = shallow(<NewBoxForm />);
  let serialized = toJson(wrapper);
  expect(serialized).toMatchSnapshot();
});
  

//testing if input changes state changes
it("allows for changes in height, width, color", function() {
 
    let wrapper = mount(<NewBoxForm />);
    const heightInput = wrapper.find("#height")
    heightInput.instance().value = 1
    heightInput.simulate("change")

    expect(wrapper.state().height).toEqual('1');

    const widthInput = wrapper.find("#width")
    widthInput.instance().value = 4
    widthInput.simulate("change")
  
    expect(wrapper.state().width).toEqual('4');

    const colorInput = wrapper.find("#color")
    colorInput.instance().value = "blue"
    colorInput.simulate("change")
  
    expect(wrapper.state().color).toEqual("blue");
  });
  // end testing inputs

// when this form is submitted, we expect that a function is fun
// we can then expect that the state changes once we fire a "change" event
it("runs a mocked fn on submit", function () {
  const submitFn = jest.fn();
  let wrapper = mount(
    <NewBoxForm addBox={submitFn} />
  );
  const form = wrapper.find("form")

  form.simulate("submit")

  expect(submitFn).toHaveBeenCalled();
});

// end mock test
// let's go and add some inputs and expect the state to reset on submit
it("resets state on submit", function () {
  const submitForm = jest.fn();
  
  let wrapper = mount(<NewBoxForm  addBox={submitForm}/>);
  const heightInput = wrapper.find("#height")
  heightInput.instance().value = 1
  heightInput.simulate("change")

  expect(wrapper.state().height).toEqual('1');

  const widthInput = wrapper.find("#width")
  widthInput.instance().value = 4
  widthInput.simulate("change")

  const colorInput = wrapper.find("#color")
  colorInput.instance().value = "blue"
  colorInput.simulate("change")

  const form = wrapper.find("form");
  form.simulate("submit");

  // after submit, we expect the state to reset
  expect(wrapper.state()).toEqual({
    "color": "",
    "height": 0,
    "width": 0
  })
});
//end reset test