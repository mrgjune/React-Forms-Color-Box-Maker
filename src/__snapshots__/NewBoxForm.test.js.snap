// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<form
  onSubmit={[Function]}
>
  <label
    htmlFor="color"
  >
    Background Color:
  </label>
  <input
    id="color"
    name="color"
    onChange={[Function]}
    value=""
  />
  <label
    htmlFor="width"
  >
    Width:
  </label>
  <input
    id="width"
    name="width"
    onChange={[Function]}
    type="number"
    value={0}
  />
  <label
    htmlFor="height"
  >
    Height:
  </label>
  <input
    id="height"
    name="height"
    onChange={[Function]}
    type="number"
    value={0}
  />
  <button>
    Add a new Box!
  </button>
</form>
`;
